web/src/pages/_app.tsx

```diff
@@ -19,18 +20,29 @@ interface Props extends AppProps {
 const queryClient = new QueryClient();
 
 function Account({ Component, pageProps: { session, ...pageProps } }: Props) {
+
+  const router = useRouter();
+  const isLogin = router.pathname == "/login"
+  const isRegister = router.pathname == "/register"
+
   return (
     <>
       <Head>
         <title>Daydream</title>
         <meta name="viewport" content="width=device-width, initial-scale=1" />
       </Head>
       <SessionProvider session={session}>
-        <QueryClientProvider client={queryClient}>
-          <TopNav />
-          <Component {...pageProps} />
-          <BottomNav />
-        </QueryClientProvider>
+        {isLogin || isRegister ? (
+          <QueryClientProvider client={queryClient}>
+            <Component {...pageProps} />
+          </QueryClientProvider>
+        ) : (
+          <QueryClientProvider client={queryClient}>
+            <TopNav />
+            <Component {...pageProps} />
+            <BottomNav />
+          </QueryClientProvider>
+        )}
```

Comment:
Ok sorry for getting this again, but you can extract the `<QueryClientProvider />` to outside the conditional since it is the same in both branches. This change will dry up the code a little

Reply:
I told Fabri to ignore any further changes on this for now since I'll refactor this soon and remove this conditional so that the navbar properly transitions in/out whenever the route changes.

Reply:
I'll do it on this branch btw, so don't approve just yet